app/controllers/application_controller.rb

```diff
@@ -19,6 +19,11 @@ class ApplicationController < ActionController::API
     render :json => error_document.to_h, :status => :bad_request
   end
 
+  rescue_from ActiveRecord::RecordNotUnique do |_exception|
+    error_document = ManageIQ::API::Common::ErrorDocument.new.add(400, "Record not unique")
```

Comment:
can we append a bit more info here as to what was the exception, i.e. to see which field was the issue.
Maybe do something similar to line 17 above and return something like "Record not unique - #{exception_msg}"
might be more useful for caller.

Reply:
We can, but there is no nice format (error doesn't contain which attribute is wrong, but what index is wrong), because it doesn't have active_record validation but database validation